### DISCUSSION TITLE
sloeber teensy issue

### DISCUSSION THREAD

**Discussion by subseapark** [Sloeber/arduino-eclipse-plugin#1453] [General]
I'm using Sloeber ver4.4 to upload code onto a Teensy module without success. This is my first time using Sloeber with a Teensy. To verify that the Teensy code was correctly installed and that the hardware was functional I ran the code on the Arduino IDE and it worked fine.
 
The following error is generated by Sloeber when I compile the code:

```
  'Starting C++ compile'
  "C:\Sloeber\arduinoPlugin\packages\arduino\hardware\teensy/../tools/precompile_helper" "C:\Sloeber\arduinoPlugin\packages\arduino\hardware\teensy\avr/cores/teensy3" "C:\Users\owner\Documents\arduino\Test_tweensy\Release" "C:\Sloeber\arduinoPlugin\packages\arduino\hardware\teensy/../tools/arm/bin/arm-none-eabi-g++" -x c++-header -Os --specs=nano.specs -g -Wall -ffunction-sections -fdata-sections -nostdlib -mno-unaligned-access -MMD -fno-exceptions -fpermissive -felide-constructors -std=gnu++14 -Wno-error=narrowing -fno-rtti -mthumb -mcpu=cortex-m0plus -fsingle-precision-constant -D__MKL26Z64__ -DTEENSYDUINO=156 -DARDUINO=10812 -DARDUINO_TEENSYLC -DF_CPU=48000000 -DUSB_SERIAL -DLAYOUT_US_INTERNATIONAL "-IC:\Sloeber\arduinoPlugin\packages\arduino\hardware\teensy\avr/cores/teensy3" "C:\Users\owner\Documents\arduino\Test_tweensy\Release/pch/Arduino.h" -o "C:\Users\owner\Documents\arduino\Test_tweensy\Release/pch/Arduino.h.gch"
  process_begin: CreateProcess(NULL, C:\Sloeber\arduinoPlugin\packages\arduino\hardware\teensy/../tools/precompile_helper C:\Sloeber\arduinoPlugin\packages\arduino\hardware\teensy\avr/cores/teensy3 C:\Users\owner\Documents\arduino\Test_tweensy\Release C:\Sloeber\arduinoPlugin\packages\arduino\hardware\teensy/../tools/arm/bin/arm-none-eabi-g++ -x c++-header -Os --specs=nano.specs -g -Wall -ffunction-sections -fdata-sections -nostdlib -mno-unaligned-access -MMD -fno-exceptions -fpermissive -felide-constructors -std=gnu++14 -Wno-error=narrowing -fno-rtti -mthumb -mcpu=cortex-m0plus -fsingle-precision-constant -D__MKL26Z64__ -DTEENSYDUINO=156 -DARDUINO=10812 -DARDUINO_TEENSYLC -DF_CPU=48000000 -DUSB_SERIAL -DLAYOUT_US_INTERNATIONAL -IC:\Sloeber\arduinoPlugin\packages\arduino\hardware\teensy\avr/cores/teensy3 C:\Users\owner\Documents\arduino\Test_tweensy\Release/pch/Arduino.h -o C:\Users\owner\Documents\arduino\Test_tweensy\Release/pch/Arduino.h.gch, ...) failed.
  make (e=2): The system cannot find the file specified.
  
  make: *** [subdir.mk:20: sloeber.ino.cpp.o] Error 2
  "C:/Sloeber/arduinoPlugin/tools/make/make all" terminated with exit code 2. Build might be incomplete.
  
  06:16:05 Build Failed. 1 errors, 0 warnings. (took 3s.451ms)

```
Any ideas how to resolve this?

**Comment by jantje** [comment]
You did not instal teensy the way it is documented
see https://eclipse.baeyens.it/how_to.shtml#/i

**Comment by subseapark** [comment]
Thanks for the reply.

 

I did however follow the instructions identified, and have repeated them again, and still no joy.

 

A screenshot of the two windows described, with the Teensy paths included, are shown below.

 

Is there anything else that needs to be done or have I misinterpreted the instructions?

 



 

 



 

Regards

MArk

**Comment by jantje** [comment]
The log you provided shows
`C:\Sloeber\arduinoPlugin\packages\arduino\hardware\teensy/.`
Which is not compliant with the content on
https://eclipse.baeyens.it/how_to.shtml#/i

> Simply install arduiono IDE 1.8 (or later) and Teensyduino 1.5 (or later) and then, inside Arduino Eclipse, go to Windows -> Arduino -> Private Hardware path. Add a new path pointing to your Teensy hardware installation directory. For windows this is something like "C:\Program Files (x86)\arduino\hardware\teensy"

**Comment by subseapark** [comment]
Would you mind expanding on your comment as the content of the instructions. With regard to the Arduino and Teensyduino versions these do indeed appear to be compliant so I m at a loss to understand the brevity of your response.

The log you provided shows
C:\Sloeber\arduinoPlugin\packages\arduino\hardware\teensy/.
Which is not compliant with the content on
https://eclipse.baeyens.it/how_to.shtml#/i

 

 

·         Arduino 1.8 or later – confirmed !

§  

·         Teensyduino 1.5 (or later) - confirmed

**Comment by subseapark** [comment]
Jantje,

 

You are an arse – and I’m mean that in the kindest possible way.  

 

I completely removed Sloeber v4.4 from my pc and started again from scratch. The result was the same.

 

Out of curiosity I removed V4.4 once again and loaded V4.1 to see if that would work – and it did. I then loaded V4.2 to see if that would work with a Teensy board and it did; without any issues whatsoever. 

 

Buoyed by this success I skipped V4.3 and once again tried a clean copy of V4.4. This gave the same error as originally reported. The evidence therefore suggests that the issue is inherent to Sloeber V4.4; and not the schoolboy error you assumed. 

 

Regards

Mark
